providers.go

```diff
@@ -0,0 +1,50 @@
+package stereoscope
+
+import (
+	containerdClient "github.com/anchore/stereoscope/internal/containerd"
+	"github.com/anchore/stereoscope/pkg/image"
+	"github.com/anchore/stereoscope/pkg/image/containerd"
+	"github.com/anchore/stereoscope/pkg/image/docker"
+	"github.com/anchore/stereoscope/pkg/image/oci"
+	"github.com/anchore/stereoscope/pkg/image/podman"
+	"github.com/anchore/stereoscope/pkg/image/sif"
+	"github.com/anchore/stereoscope/tagged"
+)
+
+const (
+	FileTag     = "file"
+	DirTag      = "dir"
+	DaemonTag   = "daemon"
+	PullTag     = "pull"
+	RegistryTag = "registry"
+)
+
+// ImageProviderConfig is the uber-configuration containing all configuration needed by stereoscope image providers
+type ImageProviderConfig struct {
+	Registry image.RegistryOptions
+	Platform *image.Platform
+}
+
+func ImageProviders(cfg ImageProviderConfig) tagged.Values[image.Provider] {
```

Comment:
Should this signature return `(tagged.Values[image.Provider], error)`? this makes usage less convenient and we do not actually return an error here in any of the code paths at present.

Reply:
No, I tried this and it makes handling errors much more confusing and defining a set of providers much more confusing. This effectively is only the glue code to provide necessary configuration to downstream functions.